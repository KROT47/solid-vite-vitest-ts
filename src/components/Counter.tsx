import { createSignal } from 'solid-js';
import './Counter.css';

export function Counter(): JSXElement {
  const [count, setCount] = createSignal(0);
  return (
    <button class="increment" onClick={(): void => { setCount(count() + 1) }}>
      Clicks: {count()}
    </button>
  );
}
