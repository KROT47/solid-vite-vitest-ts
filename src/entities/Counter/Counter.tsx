import { Button } from '@hope-ui/core';
import { createSignal } from 'solid-js';

const [count, setCount] = createSignal(0);

export function Counter(): JSXElement {
  return (
    <Button
      variant="solid"
      size="lg"
      color="red"
      _dark={{ color: 'red' }}
      onClick={(): void => {
        setCount(count() + 1);
      }}
    >
      Clicks: {count()}
    </Button>
  );
}
