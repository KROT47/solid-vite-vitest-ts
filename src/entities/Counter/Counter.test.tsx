import { fireEvent, render } from 'solid-testing-library';
import { describe, it } from 'vitest';

import { Counter } from '~/entities/Counter';

describe('<Counter />', () => {
  it('renders', ({ expect }) => {
    const { container, unmount } = render(() => <Counter />);
    expect(container).toMatchSnapshot();
    unmount();
  });

  it('increments value', async ({ expect }) => {
    const { queryByText, queryByTestId, unmount } = render(() => <Counter />);
    const incButton = queryByText('Increment');
    const decButton = queryByText('Decrement');
    const stateLabel = queryByTestId('counter-state');

    expect(incButton).toBeInTheDocument();
    expect(stateLabel).toHaveTextContent(/State: 0/);

    fireEvent.click(incButton);
    expect(stateLabel).toHaveTextContent(/State: 1/);

    fireEvent.click(decButton);
    expect(stateLabel).toHaveTextContent(/State: 0/);

    unmount();
  });
});
