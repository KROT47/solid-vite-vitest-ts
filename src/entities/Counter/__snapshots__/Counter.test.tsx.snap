// Vitest Snapshot v1

exports[`<Counter /> > renders 1 1`] = `
<div>
  <button
    class="hope-c-PJLV hope-c-PJLV-ictfuVy-css hope-Button-root hope-c-gcZMyU hope-c-eVLFDE hope-c-hVhtWI"
    type="button"
  >
    Clicks: 
    1
  </button>
</div>
`;
